core/interop/pinvoke/marshalling/src/MarshallingSample/MarshallingSampleNative.cs

```diff
@@ -0,0 +1,130 @@
+﻿using System;
+using System.Runtime.InteropServices;
+
+namespace MarshallingSample
+{
+    internal static class MarshallingSampleNative
+    {
+        // String marshalling APIs
+        [DllImport(nameof(MarshallingSampleNative), ExactSpelling = true, CharSet = CharSet.Ansi)]
+        public static extern int CountUtf8Characters(string value);
+
+        [DllImport(nameof(MarshallingSampleNative), ExactSpelling = true, CharSet = CharSet.Unicode)]
+        public static extern int CountUtf16Characters(string value);
+
+#if WINDOWS
```

Comment:
You might want to comment about CharSet options here. I would at least indicate the default encoding on Windows and what `Ansi` means on non-Windows.